### DISCUSSION TITLE
Add amine properties to the supcrtbl database

### DISCUSSION THREAD

**Discussion by armahapa** [reaktoro/reaktoro#422] [General]
I was trying to add MEA and the relevant ions in the superb database. For that, I downloaded the database and added a section for a few new species as below (and renamed as supcrtbl_custom.yaml):
MEA(aq):
    Name: MEA(aq)
    Formula: C2H7NO
    Elements: 1:N 1:O 2:C 7:H
    Charge: 0.0
    AggregateState: Aqueous
    StandardThermoModel:
      HKF:
        Gf: -552468.90
        Hf: -552371.00
        Sr: 297.37
        a1: 1.48909e-05
        a2: -616.3
        a3: 0.0002274
        a4: -113470.0
        c1: 30.9616
        c2: -74935.0
        wref: 80630.0
        charge: 0.0
  MEAH+:
    Name: MEAH+
    Formula: C2H8NO+
    Elements: 1:N 1:O 2:C 8:H
    Charge: 1.0
    AggregateState: Aqueous
    StandardThermoModel:
      HKF:
        Gf: -553908.0
        Hf: -553816.1
        Sr: 296.67
        a1: 1.48909e-05
        a2: -616.3
        a3: 0.0002274
        a4: -113470.0
        c1: 30.9616
        c2: 74935.0
        wref: 186630.0
        charge: 1.0

  MEACOO-:
    Name: MEACOO-
    Formula: C3H7NO2-
    Elements: 1:N 2:O 3:C 7:H
    Charge: -1.0
    AggregateState: Aqueous
    StandardThermoModel:
      HKF:
        Gf: -1046868.0
        Hf: -1046754.0
        Sr: 365.33
        a1: 1.48909e-05
        a2: -616.3
        a3: 0.0002274
        a4: -113470.0
        c1: 30.9616
        c2: -74935.0
        wref: 80630.0
        charge: -1.0
    MEACOOH(aq):
    Name: MEACOOH(aq)
    Formula: C3H8NO2
    Elements: 1:N 2:O 3:C 8:H
    Charge: 0.0
    AggregateState: Aqueous
    StandardThermoModel:
      HKF:
        Gf: -1047795.0
        Hf: -1047679.0
        Sr: 374.73
        a1: 1.48909e-05
        a2: -616.3
        a3: 0.0002274
        a4: -113470.0
        c1: 30.9616
        c2: -74935.0
        wref: 80630.0
        charge: 0.0

I loaded the custom yaml file as:
db = Database.fromFile("supcrtbl_custom.yaml")

then, I considered a system with:
system = ChemicalSystem(db,
    AqueousPhase("CO2(aq) H2O(aq) HCO3- MEA(aq) MEAH+ MEACOO-").set(ActivityModelHKF())
)

When I calculate properties, I notice that activity coefficients always come to 1 for charge-free species (MEA(aq), MEACOOH(aq)), irrespective of any input values for a's,c's, and wref. Many other electrolytes, like KOH(aq), follow the same trend. However, when I calculate the activity coefficient for CO2, it's not 1, returning a realistic value. 

Here are my questions:
a. Is this the correct way to add new species in the database when I want to calculate activity coefficients? If not, what are the other alternatives?
b. When I add a new species, do I need to add reactions somewhere in the database? like the example in the tutorial:
FormationReaction:
      Reactants: 1:CO3-2 2:H+ -1:H2O
c. Hardcoded properties: I see there is an option for hardcoded properties like:
  Ca+2:
    Name: Ca+2
    Formula: Ca+2
    Elements: 1:Ca
    Charge: 2.0
    AggregateState: Aqueous
    StandardThermoModel:
      Interpolation:
        Temperatures: [298.15, 350.0, 400.0]  # in K
        Pressures: [1.0e+5, 10.0e+5]          # in Pa
        G0: [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]
        H0: [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]
        
        Can I do the same for the activity coefficients? What would be the name of the variable?

Thanks in advance.

**Comment by allanleal** [comment]
Hi @armahapa,

Have you managed to resolve this issue in the meantime? Just to clarify, the thermodynamic database entries above do not influence the activity model. The HKF parameters you list above are used to calculate the standard Gibbs energies of the species, not their activity coefficients.

However, the HKF activity model (distinct from the HKF standard-state thermodynamic model) does not know how to compute the activity coefficients for the newly introduced neutral species and that's why they are one.

I recommend using the Extended UNIQUAC activity model in Reaktoro. You'll need to set up a database with the species of interest and a parameter file containing the necessary parameters for the Extended UNIQUAC model. While Reaktoro does not currently include these files, relevant data can be found in the literature. I suggest looking into publications by Philip Loldrup Fosbøl and Kaj Thomsen (DTU), which rely on an Extended UNIQUAC model compatible with Reaktoro's implementation. They have performed parameter fitting in the past for MEA systems.

On another note, I'm curious. Where did you take from the HKF parameters for the MEA species mentioned above?

**Reply by armahapa** [reply]
Thank you for your response and the detailed clarification. I am considering using the Supcrt database due to its extensive variety and the number of species it includes. I noticed that most species utilize the HKF thermodynamic model within the Supcrt database. However, when attempting to use the UNIQUAC model, I found that the existing UNIQUAC database provides most thermodynamic properties, such as enthalpy and Gibbs energy, as zero, with the exception of the activity coefficient.

I require thermodynamic properties, in addition to activity coefficients, for common ions and electrolytes (such as KOH and K+). Is it possible to define a system where certain properties (enthalpy and free energy) come from the Supcrt database and the HKF thermodynamic model, while the activity coefficients come from the UNIQUAC model?

Additionally, I would like to discuss the possibility of forming a system that includes species from two different databases. I have seen examples of adding species to a database after it has been loaded. Is it feasible to incorporate an entirely new database into an already-loaded database?

**Reply by allanleal** [reply]
Hi @armahapa - incorporating new species/reactions into existing databases can create thermodynamic/fitting inconsistencies. It is important to consider that the parameters in a database, for groups of species, may have been calibrated simultaneously against a set of experimental data, and introducing new species may affect the overall chemical behavior of the system. It is not something guaranteed to work, although it may work in some particular cases. So, I would suggest care with this approach.
